python/lsst/pipe/base/_quantumContext.py

```diff
@@ -214,25 +230,30 @@ def __init__(
             for ref in refs:
                 self.allOutputs.add((ref.datasetType, ref.dataId))
         self.outputsPut: set[tuple[DatasetType, DataCoordinate]] = set()
+        self.outputRefsPut: list[DatasetRef] = []
```

Comment:
@TallJimbo you added `ouputsPut` here as a tuple of DatasetType and DataCoordinate. I need the ref ID. I wasn't sure why you were storing a tuple for comparison rather than the ref itself so I added another item (which I should probably have as a set now I think about it but I wanted to retain order). Should I add the ref.id to your tuple? Or keep my own list?

Reply:
I needed to be able to do a set-difference with `allOutputs` (in `ctrl.mpexec.SingleQuantumExecutor`), so I just used the same type as that.  If you can keep that set-difference (or equivalent) working, feel free to change it however you'd like.